src/contracts/signature-drop.ts

```diff
@@ -463,6 +464,38 @@ export class SignatureDrop extends Erc721<SignatureDropContract> {
     return results;
   }
 
+  /**
+   * Construct a claim transaction without executing it.
+   * This is useful for estimating the gas cost of a claim transaction, overriding transaction options and having fine grained control over the transaction execution.
+   * @param destinationAddress
+   * @param quantity
+   * @param checkERC20Allowance
+   */
+  public async getClaimTransaction(
+    destinationAddress: string,
+    quantity: BigNumberish,
+    checkERC20Allowance = true, // TODO split up allowance checks
+  ): Promise<TransactionTask> {
+    const claimVerification = await this.prepareClaim(
+      quantity,
+      checkERC20Allowance,
+    );
+    return TransactionTask.builder(this.contractWrapper, "claim")
```

Comment:
**nitpick** would be more concise with object params pattern instead of factory/builder and (imo but maybe wrong) more idiomatic for JS

Reply:
Also, instead of passing `this.contractWrapper` here, could we add a function called `buildTransaction` to `contractWrapper` that lets us do `this.contractWrapper.buildTransaction` with the exact same API we've been using before for `sendTransaction`